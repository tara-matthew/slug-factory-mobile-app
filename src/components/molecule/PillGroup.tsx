import React from 'react';
import Pill from "../atom/Pill";
import {StyleSheet, View} from "react-native";

const PillGroup = ({pills}) => {
    return (
        <View className={"flex flex-row flex-wrap mb-9"} style={styles.container}>
            {pills.map((pill, index) => (
                <Pill key={index} title={pill.title} />
            ))}
        </View>
    )
}

const styles = StyleSheet.create( {
    container: {
        rowGap: 16,
        columnGap: 12
    },
})

export default PillGroup;
