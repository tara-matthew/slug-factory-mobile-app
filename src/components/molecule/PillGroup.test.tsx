import React from "react";
import PillGroup from "./PillGroup";
import { render } from "@testing-library/react-native";

const pills = [
    {
        title: "Pill 1",
    },
    {
        title: "Pill 2",
    },
    {
        title: "Pill 3",
    },
];
it("renders correctly", () => {
    const { toJSON } = render(<PillGroup pills={ pills } />);

    expect(toJSON()).toMatchSnapshot();
});

it("displays all pills with titles", () => {
    const { getByText } = render(<PillGroup pills={ pills } />);

    pills.forEach((pill) => {
        expect(getByText(pill.title)).toBeTruthy();
    });
});
