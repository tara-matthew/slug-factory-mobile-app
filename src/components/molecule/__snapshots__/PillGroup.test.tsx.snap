// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="flex flex-row flex-wrap mb-9"
  style={
    {
      "columnGap": 12,
      "rowGap": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      className="text-center text-black"
    >
      Pill 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      className="text-center text-black"
    >
      Pill 2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      className="text-center text-black"
    >
      Pill 3
    </Text>
  </View>
</View>
`;
